expat/lib/expat.h

```diff
@@ -124,9 +124,36 @@ enum XML_Error {
   XML_ERROR_RESERVED_PREFIX_XMLNS,
   XML_ERROR_RESERVED_NAMESPACE_URI,
   /* Added in 2.2.1. */
-  XML_ERROR_INVALID_ARGUMENT
+  XML_ERROR_INVALID_ARGUMENT,
+  /* Added in 2.3.0 */
+  XML_ERROR_ENTITY_TOO_LARGE,
+  XML_ERROR_ENTITY_EXPANSION_LIMIT,
+  XML_ERROR_ENTITY_RECURSION_LIMIT
 };
 
+enum XML_Options {
+  /* Added in 2.3.0 */
+  XML_OPTION_NONE = 0,
+  XML_OPTION_HUGE_ENTITES = 1<<0 /* No limits for entity expansion */
+};
+
+/* Entity expansion protection
+ *
+ * Mitigation against billion laugh and quadratic blowup attacks.
+ *
+ * XML_ENTITY_RECURSION_LIMIT confines nesting of entities within entities.
```

Comment:
I'm starting to think if "nesting" would be a better term since "recursion" seems to imply that something can reference itself (which XML entities cannot).